src/filemap.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+
+// TODO tests
+
+use strings::string_buffer::StringBuffer;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{Write, stdout};
+use WriteMode;
+use NewlineStyle;
+use config::Config;
+
+// A map of the files of a crate, with their new content
+pub type FileMap = HashMap<String, StringBuffer>;
+
+// Append a newline to the end of each file.
+pub fn append_newlines(file_map: &mut FileMap) {
+    for (_, s) in file_map.iter_mut() {
+        s.push_str("\n");
+    }
+}
+
+pub fn write_all_files(file_map: &FileMap,
+                       mode: WriteMode,
+                       config: &Config)
+                       -> Result<(HashMap<String, String>), ::std::io::Error> {
+    let mut result = HashMap::new();
+    for filename in file_map.keys() {
+        let one_result = try!(write_file(file_map, filename, mode, config));
+        if let Some(r) = one_result {
+            result.insert(filename.clone(), r);
+        }
+    }
+
+    Ok(result)
+}
+
+fn write_file(file_map: &FileMap,
+              filename: &str,
+              mode: WriteMode,
+              config: &Config)
+              -> Result<Option<String>, ::std::io::Error> {
+    let text = &file_map[filename];
```

Comment:
Is `file_map` used anywhere else in this function? If not, we may just pass the text instead, since we're already iterating over the keys in `write_all_files`. (Iterate over key-value pair with `iter`)